catalog/ui/src/app/components/ServiceNamespaceSelect.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { useMemo, useState } from 'react';
+import useSWR, { useSWRConfig } from 'swr';
+import useSession from '@app/utils/useSession';
+import { ContextSelector, ContextSelectorItem } from '@patternfly/react-core';
+import { ServiceNamespace } from '@app/types';
+import { apiPaths, fetcher } from '@app/api';
+
+import './service-namespace-select.css';
+
+const ServiceNamespaceSelect: React.FC<{
+  allowSelectAll: boolean;
+  currentNamespaceName: string;
+  isPlain: boolean;
+  isText: boolean;
+  onSelect: (namespace: ServiceNamespace) => void;
+  selectWorkshopNamespace: boolean; 
+}> = ({
+  allowSelectAll = false,
+  currentNamespaceName,
+  isPlain = false,
+  isText = false,
+  onSelect,
+  selectWorkshopNamespace=false,
+}) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const [searchValue, setSearchValue] = React.useState('');
+  const { isAdmin, serviceNamespaces: sessionServiceNamespaces, workshopNamespaces: sessionWorkshopNamespaces } = useSession().getSession();
+  const enableFetchNamespaces: boolean = isAdmin;
```

Comment:
nip: as isAdmin is known that is a boolean, typescript already infers that type, therefore is not needed to declare it again

Reply:
removed declaration of boolean type.